examples/game/breakout.rs

```diff
@@ -5,268 +5,381 @@ use bevy::{
     sprite::collide_aabb::{collide, Collision},
 };
 
-/// An implementation of the classic game "Breakout"
-const TIME_STEP: f32 = 1.0 / 60.0;
+use components::*;
+use resources::*;
+
+/// Constants that can be used to fine-tune the behavior of our game
+mod config {
+    use bevy::render::color::Color;
+    use bevy::ui::Val;
+    // TODO: add various Vec2's and Transforms to this config module for clarity and consistency
+    // Blocked on https://github.com/bitshifter/glam-rs/issues/76
+
+    pub const TIME_STEP: f32 = 1.0 / 60.0;
+    pub const BACKGROUND_COLOR: Color = Color::rgb(0.9, 0.9, 0.9);
+
+    pub const PADDLE_COLOR: Color = Color::rgb(0.5, 0.5, 1.0);
+    pub const PADDLE_SPEED: f32 = 500.0;
+
+    pub const BALL_COLOR: Color = Color::rgb(1.0, 0.5, 0.5);
+
+    pub const WALL_THICKNESS: f32 = 10.0;
+    pub const WALL_COLOR: Color = Color::rgb(0.8, 0.8, 0.8);
+
+    pub const BRICK_COLOR: Color = Color::rgb(0.5, 0.5, 1.0);
+
+    pub const SCOREBOARD_COLOR: Color = Color::rgb(0.5, 0.5, 1.0);
+    pub const SCORE_COLOR: Color = Color::rgb(1.0, 0.5, 0.5);
+    pub const SCORE_FONT_SIZE: f32 = 40.0;
+    pub const SCORE_PADDING: Val = Val::Px(5.0);
+}
+
+/// A simple implementation of the classic game "Breakout"
 fn main() {
     App::build()
         .add_plugins(DefaultPlugins)
-        .insert_resource(Scoreboard { score: 0 })
-        .insert_resource(ClearColor(Color::rgb(0.9, 0.9, 0.9)))
-        .add_startup_system(setup.system())
+        .insert_resource(ClearColor(config::BACKGROUND_COLOR))
+        // This adds the Score resource with its default values: 0
+        .init_resource::<Score>()
+        // These systems run only once, before all other systems
+        .add_startup_system(spawn_cameras.system())
+        .add_startup_system(spawn_paddle.system())
+        .add_startup_system(spawn_ball.system())
+        .add_startup_system(spawn_walls.system())
+        .add_startup_system(spawn_bricks.system())
```

Comment:
I think you went a little overboard with splitting here. Can you merge these into a single system?

Reply:
I personally disagree with this, if you look at the functions, they are all lengthy enough, putting it all into one system would make for a very long function.

Or if you are going put everything into one system, it would be a good idea to split the different spawnings into their own functions anyway.

To me it makes sense to just separate those into their own system.

And alternative alternative approach:
Approach real life code even more and actually use bundles, then you could probably make the spawning code concise enough to stick into one system. (of course you have to draw the line somewhere with an example like this, so perhaps bundles are overkill)

Reply:
In addition to @speakk's perspective, I also wanted to make this code easier to iteratively build up in tutorial-style content, or extend with new functionality.

Cleanly splitting this logic makes it easier to do both :)